### ISSUE TITLE
Finetuning fails with seg fault unless CUDA_VISIBLE_DEVICES is set

### ISSUE THREAD

**Issue by nailimixaM** [microsoft/TransformerCompression#106]
This works: `python run_finetuning.py --model microsoft/phi-2`

 but : `python run_finetuning.py --model microsoft/phi-2 --ppl-eval-dataset alpaca --ppl-eval-batch-size 16 --finetune-dataset alpaca --finetune-train-nsamples 8000 --finetune-train-seqlen 1024 --finetune-train-batch-size 3 --lora-alpha 10 --lora-dropout 0.05 --lora-r 32 --eval-steps 16 --save-steps 16 --lora-target-option attn_head_and_mlp --no-wandb` gives a segfault unless the GPU is set:

AWS dependencies are not imported since dependencies are missing. You can install them with
   pip install 'syne-tune[aws]'
or (for almost everything)
   pip install 'syne-tune[basic]'
Loading checkpoint shards: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:02<00:00,  1.13s/it]
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
trainable params: 47,185,920 || all params: 2,826,869,760 || trainable%: 1.669193277584886
  0%|                                                                                                                                   | 0/1425 [00:00<?, ?it/s]/usr/local/lib/python3.10/dist-packages/torch/utils/checkpoint.py:428: UserWarning: torch.utils.checkpoint: please pass in use_reentrant=True or use_reentrant=False explicitly. The default value of use_reentrant will be updated to be False in the future. To maintain current behavior, pass use_reentrant=True. It is recommended that you use use_reentrant=False. Refer to docs for more details on the differences between the two variants.
  warnings.warn(
/usr/local/lib/python3.10/dist-packages/torch/nn/parallel/_functions.py:68: UserWarning: Was asked to gather along dimension 0, but all input tensors were scalars; will instead unsqueeze and return a vector.
  warnings.warn('Was asked to gather along dimension 0, but all '
[db9e16e4f6f7:1730908:0:1734535] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x72006500530028)
==== backtrace (tid:1734535) ====
 0 0x0000000000042520 __sigaction()  ???:0
 1 0x00000000000406de ncclRecv()  ???:0
 2 0x0000000000045c6d pncclRedOpDestroy()  ???:0
 3 0x000000000004909d pncclRedOpDestroy()  ???:0
 4 0x000000000004a078 pncclRedOpDestroy()  ???:0
 5 0x000000000004a7ec ncclGroupEnd()  ???:0
 6 0x0000000000e0953c torch::cuda::nccl::AutoNcclGroup::~AutoNcclGroup()  :0
 7 0x0000000000e0adf9 torch::cuda::nccl::reduce()  ???:0
 8 0x0000000000c065b7 THCPModule_nccl_reduce()  ???:0
 9 0x000000000015fe38 PyObject_CallFunctionObjArgs()  ???:0
10 0x00000000001565eb _PyObject_MakeTpCall()  ???:0
11 0x000000000014f1f1 _PyEval_EvalFrameDefault()  ???:0
12 0x000000000016070c _PyFunction_Vectorcall()  ???:0
13 0x000000000014a0d1 _PyEval_EvalFrameDefault()  ???:0
14 0x000000000016070c _PyFunction_Vectorcall()  ???:0
15 0x0000000000148e0d _PyEval_EvalFrameDefault()  ???:0
16 0x000000000016070c _PyFunction_Vectorcall()  ???:0
17 0x000000000014e8a2 _PyEval_EvalFrameDefault()  ???:0
18 0x000000000016070c _PyFunction_Vectorcall()  ???:0
19 0x00000000006e79fb THPFunction_apply()  ???:0
20 0x000000000015fe38 PyObject_CallFunctionObjArgs()  ???:0
21 0x000000000016f12b PyObject_Call()  ???:0
22 0x000000000014b2c1 _PyEval_EvalFrameDefault()  ???:0
23 0x000000000016e62e PyMethod_New()  ???:0
24 0x000000000014b2c1 _PyEval_EvalFrameDefault()  ???:0
25 0x000000000016070c _PyFunction_Vectorcall()  ???:0
26 0x000000000014b2c1 _PyEval_EvalFrameDefault()  ???:0
27 0x000000000016e62e PyMethod_New()  ???:0
28 0x00000000006dfaa8 torch::autograd::PyNode::apply()  ???:0
29 0x000000000404582b torch::autograd::Node::operator()()  :0
30 0x000000000403f898 torch::autograd::Engine::evaluate_function()  ???:0
31 0x0000000004040a15 torch::autograd::Engine::thread_main()  ???:0
32 0x000000000403c60d torch::autograd::Engine::execute_with_graph_task()  ???:0
33 0x00000000006dce75 torch::autograd::python::PythonEngine::execute_with_graph_task()  ???:0
34 0x000000000403e936 torch::autograd::Engine::execute()  ???:0
35 0x00000000006dce06 torch::autograd::python::PythonEngine::execute()  ???:0
36 0x00000000006db51e THPEngine_run_backward()  ???:0
37 0x000000000015fe0e PyObject_CallFunctionObjArgs()  ???:0
38 0x00000000001565eb _PyObject_MakeTpCall()  ???:0
39 0x000000000014f908 _PyEval_EvalFrameDefault()  ???:0
40 0x000000000016070c _PyFunction_Vectorcall()  ???:0
41 0x000000000014e8a2 _PyEval_EvalFrameDefault()  ???:0
42 0x000000000016070c _PyFunction_Vectorcall()  ???:0
43 0x000000000014b2c1 _PyEval_EvalFrameDefault()  ???:0
44 0x000000000016e62e PyMethod_New()  ???:0
45 0x00000000006dfaa8 torch::autograd::PyNode::apply()  ???:0
46 0x000000000404582b torch::autograd::Node::operator()()  :0
47 0x000000000403f898 torch::autograd::Engine::evaluate_function()  ???:0
48 0x0000000004040a15 torch::autograd::Engine::thread_main()  ???:0
49 0x0000000004037dd9 torch::autograd::Engine::thread_init()  ???:0
50 0x00000000006dca01 torch::autograd::python::PythonEngine::thread_init()  ???:0
51 0x00000000000dc253 std::error_code::default_error_condition()  ???:0
52 0x0000000000094ac3 pthread_condattr_setpshared()  ???:0
53 0x0000000000125bf4 clone()  ???:0

Segmentation fault (core dumped)
